mesonbuild/interpreter.py

```diff
@@ -1462,9 +1462,25 @@ def func_configuration_data(self, node, args, kwargs):
             raise InterpreterException('configuration_data takes no arguments')
         return ConfigurationDataHolder()
 
+    def fix_docdir_default_value(self, project_name):
+        '''
+        The value of docdir is supposed to be `share/doc/pkgname`, but we can
+        only set the default values of builtin options as static strings, so
+        it's set as `share/doc`.
+        We only know the value of pkgname after parsing the name in
+        func_project. So it is only there that we can override the value of
+        docdir with the correct one.
+        '''
+        # share/doc/project-name
+        value = self.coredata.get_builtin_option('docdir')
+        # share/doc/{project_name}
+        newvalue = value.replace('project-name', project_name)
```

Comment:
It can only ever replace `project-name` in the default value set by *us* in `coredata.py` which is `usr/share/project-name`. We cannot ever replace it anywhere else, so it can't replace 'all project-name'.